blog/2022-02-24-starboard.md

```diff
@@ -0,0 +1,92 @@
+---
+title: Improving XKS security using Starboard
+description: Xenit evolves Starboard for continuous scanning of production workloads.
+authors:
+  - name: Edvin Norling
+tags:
+  - security
+  - kubernetes
+  - starboard
+  - trivy
+---
+
+How sure are you that you have close to 0 critical CVEs in your Kubernetes cluster?
+
+Just like for all companies today security is top of mind for Xenit and we try to come with a solution for this question.
+
+Today we already scan our images in our CI/CD pipeline at creation time using Trivy but what about new CVEs that get found every day?
+
+The increasing rates of cyber crime (by some measures, cyber crimes now outnumber all other crimes put together) which makes it harder for companies to protect them selfs.
+The faster we can fix relatively simple problems like patching a CVE on container level the more secure we will be.
+
+Xenit is hosting a number of Kubernetes clusters for our self and our customers and we want a quick way of visualizing CVEs on a platform and a customer level.
+Without having to jumping around to different customers and run some script to find out the answer to that question.
+
+<!-- truncate -->
+
+Could one solution be to scan the container images continuously that is running in our Kubernetes clusters?
+Or should we just continuously scan the images in our private repository? But what about the images that isn't our private repository?
+
+Do we and our customers even store all the images of our cluster in the private repositories?
+
+Since we have a central monitoring solution that we use to monitor the status of all our clusters why not think of CVEs just like another metric?
+
+To be able to solve all our questions we decided to go with continuously scanning the images that is in the Kubernetes cluster.
```

Comment:
I assume this refers to scanning the local cache of images? If so, the sentence could be clearer about that.

Reply:
No this have nothing to do with the node container images.
We look through all the deployments/jobs etc that exist in the cluster and then we scan them. So there is no node dependencies. I tried to clarify it a bit.